test/utils/svm/utils.ts

```diff
@@ -0,0 +1,359 @@
+  );
+  return { state };
+};
+
+// Enables a token route and creates the program vault ATA.
+export const enableRoute = async (
+  signer: KeyPairSigner,
+  state: Address,
+  mint: Address,
+  routeChainId: bigint,
+  solanaClient: RpcClient
+) => {
+  const [vault] = await findAssociatedTokenPda({
+    owner: state,
+    tokenProgram: TOKEN_2022_PROGRAM_ADDRESS,
+    mint,
+  });
+
+  const createAssociatedTokenIdempotentIx = getCreateAssociatedTokenIdempotentInstruction({
+    payer: signer,
+    owner: state,
+    mint,
+    ata: vault,
+    systemProgram: SYSTEM_PROGRAM_ADDRESS,
+    tokenProgram: TOKEN_2022_PROGRAM_ADDRESS,
+  });
+
+  const route = await createRoutePda(mint, 0n, routeChainId);
+  const eventAuthority = await getEventAuthority();
+
+  const input: SetEnableRouteInput = {
+    signer,
+    state,
+    vault,
+    payer: signer,
+    route,
+    originTokenMint: mint,
+    tokenProgram: TOKEN_2022_PROGRAM_ADDRESS,
+    associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ADDRESS,
+    systemProgram: SYSTEM_PROGRAM_ADDRESS,
+    eventAuthority,
+    program: address(SvmSpokeClient.SVM_SPOKE_PROGRAM_ADDRESS),
+    originToken: mint,
+    destinationChainId: 1,
+    enabled: true,
+  };
+  const setEnableRouteIx = await SvmSpokeClient.getSetEnableRouteInstructionAsync(input);
+
+  await pipe(
+    await createDefaultTransaction(solanaClient, signer),
+    (tx) => appendTransactionMessageInstruction(createAssociatedTokenIdempotentIx, tx),
+    (tx) => appendTransactionMessageInstruction(setEnableRouteIx, tx),
+    (tx) => signAndSendTransaction(solanaClient, tx)
+  );
+  return { vault, route };
+};
+
+// Executes a deposit into the SVM Spoke vault.
+export const deposit = async (
+  signer: KeyPairSigner,
+  depositData: {
+    depositor: Address;
+    recipient: Address;
+    inputToken: Address;
+    outputToken: Address;
+    inputAmount: bigint;
+    outputAmount: bigint;
+    destinationChainId: number;
+    exclusiveRelayer: Address;
+    quoteTimestamp: number;
+    fillDeadline: number;
+    exclusivityParameter: number;
+    message: ReadonlyUint8Array;
+  },
+  depositAccounts: {
+    state: Address;
+    route: Address;
+    signer: Address;
+    depositorTokenAccount: Address;
+    vault: Address;
+    mint: Address;
+    tokenProgram: Address;
+    program: Address;
+  },
+  tokenDecimals: number,
+  solanaClient: RpcClient
+) => {
+  const eventAuthority = await getEventAuthority();
+
+  const approveIx = getApproveCheckedInstruction({
+    source: depositAccounts.depositorTokenAccount,
+    mint: depositAccounts.mint,
+    delegate: depositAccounts.state,
+    owner: depositData.depositor,
+    amount: depositData.inputAmount,
+    decimals: tokenDecimals,
+  });
+
+  const depositInput: DepositInput = {
+    ...depositData,
+    ...depositAccounts,
+    eventAuthority,
+    signer,
+  };
+
+  const depositIx = await SvmSpokeClient.getDepositInstructionAsync(depositInput);
+
+  return pipe(
```

Comment:
Note: These are test-only helper functions designed to simplify interactions with the program. They do not estimate optimal compute units or configure priority fees for better block inclusion. However, they can serve as a reference for implementing the program interaction interface in the SDK.